app/client/src/components/propertyControls/IconSelectControl.tsx

```diff
@@ -131,13 +131,28 @@ class IconSelectControl extends BaseControl<
   private searchInput: React.RefObject<HTMLInputElement>;
   id: string = generateReactKey();
 
+  static defaultProps = {
+    showNoneIcon: true,
+  };
+
   constructor(props: IconSelectControlProps) {
     super(props);
     this.iconSelectTargetRef = React.createRef();
     this.virtuosoRef = React.createRef();
     this.searchInput = React.createRef();
     this.initialItemIndex = 0;
     this.filteredItems = [];
+
+    const noneIndex = ICON_NAMES.indexOf(NONE);
```

Comment:
Do we really need to check the index of `NONE` since we are unshifting it and it would always be at 0th position?
We can just use the icon set to check if `NONE` is there or not.
How about the following change?

```suggestion
    const noneIconExists = icons.has(NONE);

    if (props.showNoneIcon && !noneIconExists) {
      ICON_NAMES.unshift(NONE);
      icons.add(NONE);
    } else if (!props.showNoneIcon && noneIconExists) {
      ICON_NAMES.shift();
      icons.delete(NONE);
    }
```

Reply:
Multiple instances of the `IconSelectControl` class may be created, and each instance modifies the ICON_NAMES array and the icons set. Without this logic, the NONE icon may be added or removed multiple times, leading to unexpected behaviour. That's why you are adding and removing the NONE icon. Right?
Can you add a comment explaining the same?